### ISSUE TITLE
uninitialized variables

### ISSUE THREAD

**Issue by maddin200** [BioMedIA/MIRTK#660]
Applications/src/merge-surfaces.cc:700]  maybe uninitialized variables: cost1  cost2

**Comment by schuhschuh** [comment]
A static code analyzer may think so, indeed.

However, the variables are guaranteed to be initialized before their first use:
```
double l1, l2, cost1, cost2;
Array<Vector3D<vtkIdType>> tris1, tris2;
for (int dj = -1; dj <= 1; dj += 2) {
  [...]
  auto &cost = (dj == -1 ? cost1 : cost2);
  cost = 0., [...];
  for (int iter = 0; iter < max_iter; ++iter) {
    [...]
  }
}

for (auto &tri : (cost1 <= cost2 ? tris1 : tris2)) {
  [...]
}
```

The first for loop will be iterated exactly twice. Upon first iteration `dj == -1` and `cost1` will be set to `0` before inner-most for loop. During the second iteration, `dj == 1` and `cost2` will be used and initialized to `0` first. Both via the `cost` reference inside the loop which is set based on the value of the loop variable `dj`. The initialization with `0.` could be moved outside the for loop to avoid such oversight.

**Comment by schuhschuh** [comment]
Closed by ce3d696a45639e9d6cdf225e3a574768c44b9d16.
